### DISCUSSION TITLE
Error when render primevue chart

### DISCUSSION THREAD

**Discussion by ABDALLAHMETWALLY250** [primefaces/community#830] [PrimeVue]
<template>
  <div class="row my-4 border p-3 rounded-2 me-3">
    <div class="title d-flex align-items-center justify-content-between">
      <div class="d-flex align-items-center justify-content-start flex-column">
        <h5 class="text-point mb-0 w-100">{{ $t("Total profits") }}</h5>
        <span class="points mb-0 w-100">{{ total_profit }} </span>
      </div>
      <div class="form-group w-25">
        <Multiselect
          v-model="nationalityValue_total_profit"
          :options="nationalities"
          :placeholder="$t('choose_nationality')"
          label="title"
          track-by="id"
          :clear-on-select="true"
          select-label=""
          :hide-selected="true"
          @keyup="searchNationality"
          @update:model-value="profitss(nationalityValue_total_profit.id)"
        >
        </Multiselect>
      </div>
    </div>

    <div class="card">
      <Chart
        type="bar"
        :data="chartDataTwoTotal"
        :options="chartOptionsTwoTotal"
        class="Books"
      />
    </div>
  </div>
</template>

<script>
  import Chart from "primevue/chart";
  import Multiselect from "vue-multiselect";
  import axios from "axios";
  export default {
    data() {
      return {
        chartDataTwoTotal: null,
        chartOptionsTwoTotal: null,
        nationalities: [],
        nationalityValue_total_profit: "",
        total_profit: 0,
        months: [],
        profitssss: [],
        labels_profits: [],
        labels_months: [],
      };
    },
    mounted() {
      this.chartDataTwoTotal = this.setchartDataTwoTotal();
      this.chartOptionsTwoTotal = this.setchartOptionsTwoTotal();
      this.profitss();
    },
    methods: {
      searchNationality(event) {
        const searchText = {
          word: event == undefined ? "" : event.target.value,
        };
        axios
          .post(`/fetch_custom_countries`, searchText)
          .then((res) => {
            // console.log(res.data.data.data, "searchNationality");
            this.nationalities = res.data.data.data;
          })
          .catch((err) => console.log(err));
      },
      setchartDataTwoTotal() {
        const documentStyle = getComputedStyle(document.documentElement);
        return {
          labels: this.months,
          datasets: [
            {
              type: "line",
              label: this.$t("Profit rate"),
              borderColor: documentStyle.getPropertyValue("--blue-500"),
              borderWidth: 2,
              fill: false,
              tension: 0.4,
              data: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 10499.68, 12935.09],
            },
            {
              type: "bar",
              label: this.$t("required_percentage"),
              backgroundColor: documentStyle.getPropertyValue("--orange-500"),
              data: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 10499.68, 12935.09],
            },
          ],
        };
      },
      setchartOptionsTwoTotal() {
        const documentStyle = getComputedStyle(document.documentElement);
        const textColor = documentStyle.getPropertyValue("--text-color");
        const textColorSecondary = documentStyle.getPropertyValue(
          "--text-color-secondary"
        );
        const surfaceBorder =
          documentStyle.getPropertyValue("--surface-border");
        return {
          maintainAspectRatio: false,
          aspectRatio: 0.6,
          plugins: {
            legend: {
              labels: {
                color: textColor,
              },
            },
          },
          scales: {
            x: {
              ticks: {
                color: textColorSecondary,
              },
              grid: {
                color: surfaceBorder,
              },
            },
            y: {
              ticks: {
                color: textColorSecondary,
              },
              grid: {
                color: surfaceBorder,
              },
            },
          },
        };
      },
      async profitss(country_id) {
        const formData = new FormData();
        formData.append("country_id", country_id ? country_id : 598);
        const res = await axios.post("/booking_profit_chart", formData);
        this.labels_months = res.data.data.months;
        this.labels_profits = res.data.data.profits;
        this.total_profit = res.data.data.total_profit;
        this.labels_months.map((label) => {
          this.months.push(label);
        });
        this.labels_profits.map((prof) => {
          this.profitssss.push(prof);
        });
        console.log(this.profitssss, "profitssss");
      },
    },
    components: {
      Chart,
      Multiselect,
    },
  };
</script>

<style lang="scss" scoped>
  .text_point {
    color: #656b78;
    font-family: "regular";
    font-size: 0.8rem;
    margin: 0.5rem 0 0.5rem 0;
  }
  .bg-img {
    background: #0a909b;
  }
  .point {
    color: #0b1427;
    font-weight: bold;
    font-size: 1.3rem;
    display: flex;
    align-content: center;
    justify-content: flex-end;
    flex-direction: row-reverse;
  }
  .p-chart.h-80 {
    min-height: 28dvh;
  }
  .linechart {
    min-height: 28dvh;
  }
  .text-book {
    color: #0b1427;
    font-weight: bold;
    font-size: 1.2rem;
  }
  .Books {
    min-height: 28dvh;
    max-height: 28dvh;
    overflow-y: auto;
  }
  .text-point {
    color: #656b78;
    font-family: "regular";
    font-size: 1rem;
  }
  .points {
    color: #0b1427;
    font-size: 2rem;
    font-family: "bold";
  }
  thead {
    background-color: #eef4f4 !important;
    th.text-center {
      padding: 0.8rem;
    }
  }
  .pendding {
    color: #c57016;
  }
  .cancel {
    color: #e23535;
  }
  .accept {
    color: #2a8f6f;
  }
  .initial {
    color: #1664b8;
  }
</style>

Error is : Maximum call stack size exceeded
RangeError: Maximum call stack size exceeded
    at Object.get2 [as get] (webpack-internal:///./node_modules/@vue/reactivity/dist/reactivity.esm-bundler.js:466:23)
    at toRaw (webpack-internal:///./node_modules/@vue/reactivity/dist/reactivity.esm-bundler.js:919:35)
    at instrumentations.<computed> (webpack-internal:///./node_modules/@vue/reactivity/dist/reactivity.esm-bundler.js:453:19)
    at Proxy.value (webpack-internal:///./node_modules/chart.js/dist/chunks/helpers.segment.mjs:500:26)
    at instrumentations.<computed> (webpack-internal:///./node_modules/@vue/reactivity/dist/reactivity.esm-bundler.js:453:36)
    at Proxy.value (webpack-internal:///./node_modules/chart.js/dist/chunks/helpers.segment.mjs:500:26)
    at instrumentations.<computed> (webpack-internal:///./node_modules/@vue/reactivity/dist/reactivity.esm-bundler.js:453:36)
    at Proxy.value (webpack-internal:///./node_modules/chart.js/dist/chunks/helpers.segment.mjs:500:26)
    at instrumentations.<computed> (webpack-internal:///./node_modules/@vue/reactivity/dist/reactivity.esm-bundler.js:453:36)
    at Proxy.value (webpack-internal:///./node_modules/chart.js/dist/chunks/helpers.segment.mjs:500:26)

**Comment by tugcekucukoglu** [comment]
Could you provide a stackblitz demo?
